### ISSUE TITLE
[java] UselessPureMethodCall false positive on ZipInputStream::getNextEntry

### ISSUE THREAD

**Issue by Jallibad** [pmd/pmd#6060]
**Affects PMD Version:** 7.17.0

**Rule:** [UselessPureMethodCall](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#uselesspuremethodcall)

**Description:**

**Code Sample demonstrating the issue:**

```java
var zipInputStream = new ZipInputStream(new ByteArrayInputStream(new byte[] {}));
zipInputStream.getNextEntry();
```

**Expected outcome:**

PMD reports a violation at line 2, but that's wrong because `getNextEntry` has a side effect of incrementing to the next entry.

**Running PMD through:** *Gradle*
